업로드된 AudioFile 관련 예외 처리 메시지 리팩토링

Johyejin-119 opened the pull request:
## Related Issues
+ solved #85

<br>

## What does this PR do?
 + [ ] 업로드된 AudioFile을 저장하지 못할 때 발생하는 예외를 custom하고 controller advice에서 handling
 + [ ] 기존 AudioFIle을 Binary 타입으로 읽어들이지 못할 때 발생하는 예외를 custom하고 controller advice에서 handling

<br>

## Solved Problem (Optional)
 +

Johyejin-119 committed:
refactor: #85 업로드된 음성 파일을 저장할 때 발생하는 오류 메시지 수정

Johyejin-119 committed:
Merge branch 'develop' of https://github.com/HiAvatar/backend into feat-johyejin

Johyejin-119 committed:
refactor: #85 음성 파일을 Binary 타입으로 읽어들일 때 발생하는 오류 메시지 추가

minoflower31 commented:
혜진님 리팩토링하신 코드 잘 봤습니다. java8 문법에 맞게 try 안에 코드를 넣어서 처리해주셨군요!
예외처리 또한 훌륭한 코드이지만 두 메서드가 처리하는 예외가 모두 RuntimeException으로 처리되고 있습니다. 개인적으로 Controller에서 파생되는 예외는 ControllerAdvice에서 처리하면 좋을 것 같은데, 만약 ControllerAdvice로 처리하게 된다면 각각 두 가지의 예외를 처리하기 위해 사용자 정의 Exception을 사용해야 한다는 번거로움이 있겠지만 더 명시적이고, ControllerAdvice에서 예외가 관리되기 때문에 에러 메시지를 보다 효율적으로 관리할 수 있다고 생각합니다.

그래서 다음과 같이 변경해주시면 더 좋은 코드가 될 것이라고 생각합니다.
1. 업로드된 AudioFile을 저장하지 못했을 때, Binary 타입으로 읽어들이지 못할 때에 대한 사용자 정의 Exception 생성하기
2. ControllerAdvice에서 예외 처리에 대한 구문을 작성 및 관리하기

너무 오지랖(?)같게 느껴지실 수 있겠지만 이런 사소함이 하나하나씩 쌓여서 더 좋은 코드를 만들어 내기 위해 함께 고민하며 실력이 늘지 않을까 싶네요ㅎㅎ

Johyejin-119 committed:
feat: #85 업로드된 음성 파일을 생성하지 못할 때 사용되는 custom exception

Johyejin-119 committed:
feat: #85 생성된 파일을 Binary 타입으로 읽어들일 수 없을 때 발생하는 custom exception

Johyejin-119 committed:
feat: #85 custom한 파일 관련 예외를 advice 클래스에서 handling

- NoCreateUploadAudioFileException: 업로드된 음성 파일을 생성하지 못할 때 발생하는 예외
- NoGetAudioFileBinaryException: 생성된 파일을 Binary 타입으로 읽어들이지 못할 때 발생하는 예외

Johyejin-119 committed:
feat: #85 파일 관련 예외처리를 RuntimeException에서 custom한 클래스로 변경

- saveUploadAudioFile(): NoCreateUploadAudioFileException 사용
- getAudioFileBinary(): NoGetAudioFileBinaryException 사용

Johyejin-119 commented:
민호님이 피드백 해주신 내용을 바탕으로, custom한 exception 클래스를 advice에서 handling하는 방식으로 예외를 처리하였습니다. custom하여 사용하니 보다 직관적으로 예외 상황을 파악할 수 있는 것 같습니다.

이번 기회에 '더 올바르고 효율적인 예외처리란 무엇인가?' 에 대해 많이 고민하고 공부할 수 있었어요. 
다시 한 번 감사드립니다:)
